modules/ballerina-mime/src/main/ballerina/ballerina/mime/natives.bal

```diff
@@ -0,0 +1,291 @@
+@Return {value:"return resulting encoded string"}
+public native function <MimeBase64Encoder encoder> encodeString (string content, string charset) (string);
+
+@Description {value:"Decode byte array using MIME Base64 encoding scheme."}
+@Param {value:"encoder: Represent MIME specific base64 decoder"}
+@Param {value:"content: the byte array to decode"}
+@Return {value:"return resulting decoded bytes"}
+public native function <MimeBase64Decoder decoder> decode (blob content) (blob);
+
+@Description {value:"Decode a given string using MIME Base64 decoding scheme. First the given string will be
+converted to a byte array with the given charset encoding. If the charset given is null default 'UTF-8' will be used.
+ Then that byte array will be decoded using MIME Base64 decoding scheme and a new string will be constructed with the
+  given charset."}
+@Param {value:"encoder: Represent MIME specific base64 decoder"}
+@Param {value:"content: string to decode"}
+@Param {value:"charset: charset used in the given string and the resulting string"}
+@Return {value:"return resulting decoded string"}
+public native function <MimeBase64Decoder decoder> decodeString (string content, string charset) (string);
+
+@Description {value:"Utility function used internally to get the bytes from a given channel."}
+@Param {value:"channel: A ByteChannel"}
+@Return {value:"return a byte array"}
+function readAll (io:ByteChannel channel) (blob) {
+    blob bytes;
+    int numberOfBytesRead;
+    bytes, numberOfBytesRead = channel.readAllBytes();
+    return bytes;
+}
+
+public const string APPLICATION_FORM_URLENCODED = "application/x-www-form-urlencoded";
+public const string APPLICATION_JSON = "application/json";
+public const string APPLICATION_OCTET_STREAM = "application/octet-stream";
+public const string APPLICATION_SVG_XML = "application/svg+xml";
+public const string APPLICATION_XHTML_XML = "application/xhtml+xml";
+public const string APPLICATION_XML = "application/xml";
+public const string MULTIPART_FORM_DATA = "multipart/form-data";
+public const string TEXT_HTML = "text/html";
+public const string TEXT_PLAIN = "text/plain";
+public const string TEXT_XML = "text/xml";
+
+
+@Description {value:"Key name for 'boundary' parameter in MediaType. This is needed for composite type media types"}
+public const string BOUNDARY = "boundary";
+
+@Description {value:"Key name for 'start' parameter in MediaType. This determines which part in the multipart message
+contains the payload"}
+public const string START = "start";
+
+@Description {value:"Key name for 'type' parameter in MediaType. This indicates the MIME media type of the 'root'
+body part"}
+public const string TYPE = "type";
+
+@Description {value:"Key name for 'charset' parameter in MediaType. Indicate the character set of the body text"}
+public const string CHARSET = "charset";
+
+public const string DEFAULT_CHARSET = "UTF-8";
+
+public const string READ_PERMISSION = "r";
```

Comment:
Do these last 2 constants need to be public? If so, need to add their `@Description` annotations.

Reply:
Fixed.